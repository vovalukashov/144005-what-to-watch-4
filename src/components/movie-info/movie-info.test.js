import React from "react";
import renderer from "react-test-renderer";
import MovieInfo from "./movie-info.jsx";

const movie = {
  name: `Сияние`,
  genre: `Drama, Horror`,
  release: 1980,
  rating: 8.4,
  ratingText: `Very good`,
  ratingCount: 857784,
  text: `A family heads to an isolated hotel for the winter where a sinister presence influences the father into violence, while his psychic son sees horrific forebodings from both past and future.`,
  director: `Stanley Kubrick`,
  starring: [`Jack Nicholson`, `Shelley Duvall`, `Danny Lloyd`],
  image: `https://resizing.flixster.com/hfmSxiWV1FlY5vtodiys3hYXPic=/fit-in/200x296.2962962962963/v1.bTsxMTE2ODAzMjtqOzE4NTM0OzEyMDA7ODAwOzEyMDA`,
  poster: `https://images-na.ssl-images-amazon.com/images/I/81nwnHTcV2L._AC_SY679_.jpg`,
  overlay: `https://cdn.shopify.com/s/files/1/1416/8662/products/shining_1980_photobusta_6_original_film_art_2000x.jpg?v=1569118744`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  runTime: `2h 26min`,
  reviews: [
    {
      textReview: `Even though The Shining is over a quarter of a century old, I challenge anyone to not get freaked out by Jack Nicholson's descent into madness. This is a rare example of something so unique that no one has been able to rip it off; instead it has been referenced time and again in pop culture.`,
      authorReview: `yancyscott1`,
      dateReview: `9 April 2007`,
      ratingReview: 9
    },
    {
      textReview: `Chilling, majestic piece of cinematic fright, this film combines all the great elements of an intellectual thriller, with the grand vision of a director who has the instinctual capacity to pace a moody horror.`,
      authorReview: `FlickJunkie`,
      dateReview: `24 July 2001`,
      ratingReview: 9
    },
  ],
};

const movies = [
  {
    name: `Сияние`,
    genre: `Drama, Horror`,
    release: 1980,
    rating: 8.4,
    ratingText: `Very good`,
    ratingCount: 857784,
    text: `A family heads to an isolated hotel for the winter where a sinister presence influences the father into violence, while his psychic son sees horrific forebodings from both past and future.`,
    director: `Stanley Kubrick`,
    starring: [`Jack Nicholson`, `Shelley Duvall`, `Danny Lloyd`],
    image: `https://resizing.flixster.com/hfmSxiWV1FlY5vtodiys3hYXPic=/fit-in/200x296.2962962962963/v1.bTsxMTE2ODAzMjtqOzE4NTM0OzEyMDA7ODAwOzEyMDA`,
    poster: `https://images-na.ssl-images-amazon.com/images/I/81nwnHTcV2L._AC_SY679_.jpg`,
    overlay: `https://cdn.shopify.com/s/files/1/1416/8662/products/shining_1980_photobusta_6_original_film_art_2000x.jpg?v=1569118744`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: `2h 26min`,
    reviews: [
      {
        textReview: `Even though The Shining is over a quarter of a century old, I challenge anyone to not get freaked out by Jack Nicholson's descent into madness. This is a rare example of something so unique that no one has been able to rip it off; instead it has been referenced time and again in pop culture.`,
        authorReview: `yancyscott1`,
        dateReview: `9 April 2007`,
        ratingReview: 9
      },
      {
        textReview: `Chilling, majestic piece of cinematic fright, this film combines all the great elements of an intellectual thriller, with the grand vision of a director who has the instinctual capacity to pace a moody horror.`,
        authorReview: `FlickJunkie`,
        dateReview: `24 July 2001`,
        ratingReview: 9
      },
    ],
  },
  {
    name: `Берегись автомобиля`,
    genre: `Drama, Comedy`,
    release: 1966,
    rating: 8.2,
    ratingText: `Very good`,
    ratingCount: 116743,
    text: `This amazing story may well be considered fantastic. A humble and shy insurance agent, amateur theater actor Yuri Detochkin is an implacable fighter for justice.`,
    director: `Eldar Ryazanov`,
    starring: [`Innocent Smoktunovsky`, `Oleg Efremov`, `Liubov Dobrzhanskaya`, `Anatoly Papanov`],
    image: `https://encrypted-tbn1.gstatic.com/images?q=tbn:ANd9GcQLEjL1jwN96kDrqosLwz4q1Zr35469CecN5KKHLkqpebEj4W3i`,
    poster: `https://avatars.mds.yandex.net/get-kinopoisk-image/1704946/059f510a-1fc7-49a6-bd73-87dc11095516/x1000`,
    overlay: `https://upload.wikimedia.org/wikipedia/ru/0/04/1966_beregis_avtomobilya.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: `1h 34min`,
    reviews: [
      {
        textReview: `Great movie, with great golden cast. This film bifurcated me in two. The komunyaga and my personal capitalist fought in me. And it’s hard for me to say which of them won. I was worried about all the characters, they were all close to me and loved.`,
        authorReview: `Alail`,
        dateReview: `27 August 2018`,
        ratingReview: 9
      },
      {
        textReview: `This is not a film, but a phenomenon. It describes only a fraction of the whole paradox of Soviet life, which has echoed to us.`,
        authorReview: `Kessid`,
        dateReview: `28 May 2018 `,
        ratingReview: 9
      },
    ],
  },
];

describe(`MovieInfo snapshot`, () => {
  it(`MovieInfo renderer`, () => {
    const handleMovieCardClick = jest.fn();
    const tree = renderer
      .create(<MovieInfo movie={movie} movies={movies} onMovieCardClick={handleMovieCardClick} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
