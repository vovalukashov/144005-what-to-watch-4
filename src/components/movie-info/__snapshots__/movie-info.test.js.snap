// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo snapshot MovieInfo renderer 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Сияние"
          src="https://cdn.shopify.com/s/files/1/1416/8662/products/shining_1980_photobusta_6_original_film_art_2000x.jpg?v=1569118744"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="main.html"
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="img/avatar.jpg"
              width="63"
            />
          </div>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Сияние
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama, Horror
            </span>
            <span
              className="movie-card__year"
            >
              1980
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="add-review.html"
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Сияние poster"
            height="327"
            src="https://images-na.ssl-images-amazon.com/images/I/81nwnHTcV2L._AC_SY679_.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              8.4
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Very good
              </span>
              <span
                className="movie-rating__count"
              >
                857784
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              A family heads to an isolated hotel for the winter where a sinister presence influences the father into violence, while his psychic son sees horrific forebodings from both past and future.
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                Stanley Kubrick
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                Jack Nicholson, Shelley Duvall, Danny Lloyd
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Fantastic Beasts: The Crimes of Grindelwald"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Bohemian Rhapsody"
              height="175"
              src="img/bohemian-rhapsody.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Macbeth"
              height="175"
              src="img/macbeth.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Aviator"
              height="175"
              src="img/aviator.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Aviator
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
